library/core/functions.compatibility.php

```diff
@@ -452,51 +452,55 @@ function valr($key, $collection, $default = false) {
     }
 }
 
-/**
- * Set a key to a value in a collection.
- *
- * Works with single keys or "dot" notation. If $key is an array, a simple
- * shallow array_merge is performed.
- *
- * @param string $key The key or property name of the value.
- * @param array &$collection The array or object to search.
- * @param mixed $value The value to set.
- * @return mixed Newly set value or if array merge.
- */
-function setvalr($key, &$collection, $value = null) {
-    if (is_array($key)) {
-        $collection = array_merge($collection, $key);
-        return null;
-    }
-
-    if (strpos($key, '.')) {
-        $path = explode('.', $key);
-
-        $selection = &$collection;
-        $mx = count($path) - 1;
-        for ($i = 0; $i <= $mx; ++$i) {
-            $subSelector = $path[$i];
+if (!function_exists('setvalr')) {
```

Comment:
I don't see how this `setvalr` function is related to `queue` implementation. 
Do I miss something?